src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -342,7 +342,20 @@ pListItem = do
                                 [Span (ident, [], []) ils] : xs)
                            _ -> B.divWith (ident, [], []) bs
   maybe id addId (lookup "id" attr) <$>
-    pInTags "li" block
+    pInTags "li" (pLabel' <|> block)
+  where 
+    pLabel' :: PandocMonad m => TagParser m Blocks
+    pLabel' = try $ pInTag TagsRequired "label" pInput'
+    pInput' :: PandocMonad m => TagParser m Blocks
+    pInput' = do
+          TagOpen _ attr' <- lookAhead $ pSatisfy $ matchTagOpen "input" [("type","checkbox")]
```

Comment:
I wasn't sure why you wanted `lookAhead` here?

Reply:
I wasn't sure if I needed to parse the closing tag too using the approach I took, so I just looked ahead and then parsed the whole thing later.

But it does make much better sense to consume the opening tag as I read it.

I'll change it, thanks!